ocagent.go

```diff
@@ -375,14 +371,57 @@ func (ae *Exporter) Stop() error {
 	return err
 }
 
+// ExportSpan exports a single span to the configured destination.
+// This is usually used by the client libraries to export to the local
+// OC agent
 func (ae *Exporter) ExportSpan(sd *trace.SpanData) {
 	if sd == nil {
 		return
 	}
 	_ = ae.traceBundler.Add(sd, 1)
 }
 
+// ExportTraceServiceRequest exports a span batch using streaming or unary gRPC depending on
+// whether `WithUnaryTraceExporter()` was used or not.
 func (ae *Exporter) ExportTraceServiceRequest(batch *agenttracepb.ExportTraceServiceRequest) error {
+	if ae.useUnaryBatchExporter {
+		return ae.exportTraceServiceRequestUnary(batch)
+	}
+	return ae.exportTraceServiceRequestStream(batch)
+}
+
+func (ae *Exporter) exportTraceServiceRequestUnary(req *agenttracepb.ExportTraceServiceRequest) error {
+	if req == nil || len(req.Spans) == 0 {
+		return nil
+	}
+
+	select {
+	case <-ae.stopCh:
+		return errStopped
+
+	default:
+		if lastConnectErr := ae.lastConnectError(); lastConnectErr != nil {
+			return fmt.Errorf("ExportTraceServiceRequest: no active connection, last connection error: %v", lastConnectErr)
```

Comment:
Since you're touching this code, could you please move the `fmt.Errorf` to where the last connection error is set? This way we avoid this formatting code in a possible hot path.

Reply:
Thanks. Will do if/when we move forward with up-streaming unary.